modules/aws/eks/eks.tf

```diff
@@ -51,6 +54,63 @@ resource "aws_eks_cluster" "this" {
     }
   }
 
+  tags       = local.global_tags
+  depends_on = [aws_cloudwatch_log_group.this]
+}
+
+
+
+# CloudWatch Encrytion
+resource "aws_kms_key" "cloudwatch_encryption" {
+  description         = "Used for cloudwatch encryption"
+  enable_key_rotation = true
+
+  policy = <<EOF
```

Comment:
This is a ugly workaround by AWS since you can't use a custom IAM role in your cloudWatch and you can't create the cloudWatch without having the IAM rule. 
But this only access for a specific cloudWatch group to the encryption from the specified ArnEquals.

This is the main reason why I use one KMS per EKS instance instead of storing it in global. If you think it's worth to move it back to global I will do that.